deploy/eck-elasticsearch/templates/elasticsearch.yaml

```diff
@@ -0,0 +1,41 @@
+---
+apiVersion: elasticsearch.k8s.elastic.co/v1
+kind: Elasticsearch
+metadata:
+  name: {{ required "A name for the Elasticsearch instance is required" .Values.name }}
+  namespace: {{ .Release.Namespace }}
+  {{- if .Values.labels }}
```

Comment:
You are not including the fixed labels from the `elasticsearch.labels` helper template.